pkg/server/server.go

```diff
@@ -952,7 +998,7 @@ func (s *Server) Check(ctx context.Context, req *openfgav1.CheckRequest) (*openf
 
 	const methodName = "check"
 	resp, checkRequestMetadata, err := commands.NewCheckCommand(
-		s.datastore,
+		s.checkDatastore,
```

Comment:
This change won't work properly.  It's related to https://github.com/openfga/openfga/issues/1394.

When you inspect the value of `datastore` inside the request, the chain of objects is 

<img width="1407" alt="image" src="https://github.com/user-attachments/assets/492f2d48-a7f6-4f1a-9023-d88b4ce3a039">

the `BoundedConcurrencyTupleReader` will rate-limit reads to the cache, which will harm performance. 

The correct order should be: `cachedDs` wraps a `combinedDs` wraps a `boundedConcurrencyDs` wraps `ds`.

I think the right way to do is is to **not** have the server keep a reference to the `cachedDs` but rather the thing that we _actually_ need to share across the requests, which is the underlying cache. Then, each call to Check creates its own `datastore object` chain of objects.

Reply:
@miparnisari I'd prefer we address that separate to these changes.